aten/src/ATen/Context.cpp

```diff
@@ -59,6 +59,48 @@ void Context::setDeterministicCuDNN(bool b) {
   deterministic_cudnn = b;
 }
 
+bool Context::deterministic() const {
+  return _deterministic;
+}
+
+void Context::setDeterministic(bool b) {
+  _deterministic = b;
+}
+
+Context::ErrorLevel Context::deterministicErrorLevel() const {
+  return deterministic_error_level;
+}
+
+void Context::setDeterministicErrorLevel(Context::ErrorLevel e) {
+  deterministic_error_level = e;
+}
+
+Context::ErrorLevel Context::longToErrorLevel(long e) {
+  switch (e) {
+    case static_cast<int>(Context::ErrorLevel::None):
+    case static_cast<int>(Context::ErrorLevel::Warn):
+    case static_cast<int>(Context::ErrorLevel::Error):
+      return static_cast<Context::ErrorLevel>(e);
+    default:
+      TORCH_CHECK(false, "error level ", e, " is invalid, must be one of 0: None, 1: Warn, or 2: Error");
+  }
+}
+
+void Context::alertNotDeterministic(std::string const& caller) {
+  if (globalContext().deterministic()) {
+    auto error_level = globalContext().deterministicErrorLevel();
+    switch (error_level) {
+      case Context::ErrorLevel::Warn:
+        TORCH_WARN(caller, " is not deterministic");
```

Comment:
It would be better to have a more detailed error message explaining why this error message is happening (it's because you set the deterministic flag) and what the user should do in this situation (whether it's file a bug asking for deterministic support, or turn off the flag, or something else.)